ui/src/main/java/org/pentaho/di/ui/spoon/trans/TransGridDelegate.java

```diff
@@ -222,6 +222,11 @@ public void addTransGrid() {
     fdView.bottom = new FormAttachment( 100, 0 );
     transGridView.setLayoutData( fdView );
 
+    //change the number column ValueMetaNumber to ValueMetaString to be compatible with sub trans numbering
+    ColumnInfo numberColumn = transGridView.getNumberColumn();
+    ValueMetaInterface numberColumnValueMeta = new ValueMetaString( "#" );
+    numberColumn.setValueMeta( numberColumnValueMeta );
```

Comment:
is there a risk of sorting issues if the number is treated as a string?  E.g. '2' sorting later than '10'.